File: src/main/java/dev/oribuin/skyblock/command/impl/BorderCommand.java
package dev.oribuin.skyblock.command.impl;

import dev.rosewood.rosegarden.RosePlugin;
import dev.rosewood.rosegarden.command.framework.BaseRoseCommand;
import dev.rosewood.rosegarden.command.framework.CommandContext;
import dev.rosewood.rosegarden.command.framework.CommandInfo;
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable;
import org.bukkit.entity.Player;
import dev.oribuin.skyblock.gui.MenuProvider;
import dev.oribuin.skyblock.gui.impl.BorderGUI;

public class BorderCommand extends BaseRoseCommand {

    public BorderCommand(RosePlugin rosePlugin) {
        super(rosePlugin);
    }

    @RoseExecutable
    public void execute(CommandContext context) {
        MenuProvider.get(BorderGUI.class).open((Player) context.getSender());
    }

    @Override
    protected CommandInfo createCommandInfo() {
        return CommandInfo.builder("border")
                .permission("skyblock.border")
                .playerOnly(false)
                .build();
    }

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/command/command/BorderCommand.kt
package xyz.oribuin.skyblock.command.command

import dev.rosewood.rosegarden.RosePlugin
import dev.rosewood.rosegarden.command.framework.CommandContext
import dev.rosewood.rosegarden.command.framework.RoseCommand
import dev.rosewood.rosegarden.command.framework.RoseCommandWrapper
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable
import org.bukkit.entity.Player
import xyz.oribuin.skyblock.gui.BorderGUI
import xyz.oribuin.skyblock.util.getMenu

class BorderCommand(rosePlugin: RosePlugin, parent: RoseCommandWrapper) : RoseCommand(rosePlugin, parent) {

    @RoseExecutable
    fun execute(context: CommandContext) = this.rosePlugin.getMenu(BorderGUI::class).openMenu(context.sender as Player)

    override fun getDefaultName(): String = "border"

    override fun getDescriptionKey(): String = "command-border-description"

    override fun getRequiredPermission(): String = "skyblock.command.border"

    override fun isPlayerOnly(): Boolean = true

}